source/LocationFilter.cpp

```diff
@@ -174,6 +174,22 @@ void LocationFilter::Load(const DataNode &node)
 
 
 
+void LocationFilter::Clear()
+{
+	planets.clear();
+	attributes.clear();
+	systems.clear();
+	governments.clear();
+	center = nullptr;
+	originMaxDistance = 0;
+	notFilters.clear();
+	neighborFilters.clear();
+	outfits.clear();
+	shipCategory.clear();
+}
+
+
+
```

Comment:
Unnecessary and also error prone.

```c++
travelRestrictions = LocationFilter(child); // load
travelRestrictions = LocationFilter{}; // clear
```
These guarantee that all fields in location filter--including new ones--are reset to the default.